Sources/BrowserServicesKit/ContentBlocking/UserScripts/ContentBlockerRulesUserScript.swift

```diff
@@ -187,11 +187,13 @@ open class ContentBlockerRulesUserScript: NSObject, UserScript {
                   !isFirstParty(requestURL: trackerUrlString, websiteURL: pageUrlStr) else { return }
 
             let entity = currentTrackerData.findEntity(forHost: requestETLDp1) ?? Entity(displayName: requestETLDp1, domains: nil, prevalence: nil)
+            let isAffiliated = resolver.isPageAffiliatedWithTrackerEntity(pageUrlString: pageUrlStr, trackerEntity: entity)
+
             let thirdPartyRequest = DetectedRequest(url: trackerUrlString,
                                                     eTLDplus1: requestETLDp1,
                                                     knownTracker: nil,
                                                     entity: entity,
-                                                    state: .allowed(reason: .otherThirdPartyRequest),
+                                                    state: .allowed(reason: isAffiliated ? .ownedByFirstParty : .otherThirdPartyRequest),
```

Comment:
Can we have a test for this? I think TrackerResolver is pretty well tested so it would be good to include this case.

Reply:
Please mind that the update is not in `TrackerResolver` (as the request is not matching any tracker definition, and essentially we have a non-tracker third party request that came from affiliated domain).
I will try to extend the `ContentBlockerRulesUserScript` test where the change is.

Reply:
@bwaresiak I have updated the tests to cover this case